### DISCUSSION TITLE
Dynamic Default Filter

### DISCUSSION THREAD

**Discussion by mentiantonio** [rappasoft/laravel-livewire-tables#418] [Q&A]
Hello, 

its' possible to set the default filter to a dynamic properties? 

public array $filters = [
    'year' => 'return current year'
];

Thank you

**Comment by rappasoft** [comment]
It's all reactive, so try something like:

```
public function mount()
{
    $this->filters['year'] = date('Y');
}
```

**Reply by mentiantonio** [reply]
Great, thank you! It's working now

**Reply by juliomotol** [reply]
I've recently bumped in to this. Yes it works, but once you try to clear the filters, it reverts back to `null` instead of `date('Y')`.

Any ideas to work around this?

**Reply by rappasoft** [reply]
You can try using the new `boot` method in 2.6 that runs on every request and see if that works (untested).

**Reply by juliomotol** [reply]
`boot()` doesnt seem to work well for this.

> Runs on every request, immediately after the component is instantiated, _**but before any other lifecycle methods are called**_

This also includes any other methods as it seems. the `boot()` precedes `resetFilter()` and `removeFilter($filter)`.

In the end, I was able to workaround this by doing:

```php

public function boot()
{
    $this->filters['date'] = ($this->filters['date'] ?? null) ?: now()->format('Y-m-d');
}

public function resetFilters(): void
{
    parent::resetFilters();

    $this->filters['date'] = ($this->filters['date'] ?? null) ?: now()->format('Y-m-d');
}

public function removeFilter($filter): void
{
    parent::removeFilter($filter);

    $this->filters['date'] = ($this->filters['date'] ?? null) ?: now()->format('Y-m-d');
}
```

Very nasty, I know :smiling_face_with_tear:

**Reply by rappasoft** [reply]
What if you just wrote your own clearFilters method where you do what you need to then run the parent code?

**Reply by bagasardi20155** [reply]
Hi, can you explain how actually to set default filter dynamically ? That `mount ` function didn't work for me. Or is it my false on implementation ?

**Reply by brn-md** [reply]
I just bumped into this, and as a newbie, I would like to know as well. If someone could explain to me, I would be very much appreciated!

**Reply by qwasyx0** [reply]
This is what you are looking for, more in [docs](https://rappasoft.com/docs/laravel-livewire-tables/v3/sorting/available-methods)
```php
 public function configure(): void
    {
        $this->setDefaultSort('DATE', 'desc');
    }
```

**Reply by lrljoe** [reply]
Please open a new thread if you're having an issue, as this thread dates back quite a while, and the advice **may** not apply anymore!
